runtime/astar/src/precompiles.rs

```diff
@@ -45,6 +48,30 @@ use xcm::latest::prelude::MultiLocation;
 /// to Erc20AssetsPrecompileSet
 pub const ASSET_PRECOMPILE_ADDRESS_PREFIX: &[u8] = &[255u8; 4];
 
+/// Filter that only allows whitelisted runtime call to pass through dispatch precompile
+#[derive(Default)]
+pub struct DispatchPrecompileFilter;
+
+impl InstanceFilter<RuntimeCall> for DispatchPrecompileFilter {
+    fn filter(&self, c: &RuntimeCall) -> bool {
+        match c {
+            RuntimeCall::Utility(pallet_utility::Call::batch { calls })
+            | RuntimeCall::Utility(pallet_utility::Call::batch_all { calls }) => {
+                for call in calls {
+                    if !DispatchPrecompileFilter::default().filter(call) {
+                        return false;
+                    }
+                }
+                true
```

Comment:
There is almost always a readable function for any `for...in` loops, together with above renaming, it will be like

```rust
    // all calls are whitelisted
    calls.iter().all(|call| WhitelistedCalls::contains(call))
```

Reply:
I know but I added it to make it verbose. I have changes it acc. to your suggestion now.